convert2rhel/unit_tests/actions/conversion/update_grub_test.py

```diff
@@ -0,0 +1,187 @@
+        (8, False, "/boot/grub2/grub.cfg", 0, 0, "Successfully updated GRUB2 on the system."),
+        (7, False, "/boot/grub2/grub.cfg", 0, 1, "Couldn't install the new images with GRUB2."),
+        (7, False, "/boot/grub2/grub.cfg", 1, 1, "GRUB2 config file generation failed."),
+    ),
+)
+def test_update_grub(
+    update_grub_instance,
+    releasever_major,
+    is_efi,
+    config_path,
+    grub2_mkconfig_exit_code,
+    grub2_install_exit_code,
+    expected,
+    monkeypatch,
+    caplog,
+):
+    monkeypatch.setattr("convert2rhel.grub.get_grub_device", mock.Mock(return_value="/dev/sda"))
+    monkeypatch.setattr("convert2rhel.grub.is_efi", mock.Mock(return_value=is_efi))
+    monkeypatch.setattr(
+        "convert2rhel.systeminfo.system_info.version", namedtuple("Version", ["major"])(releasever_major)
+    )
+    run_subprocess_mocked = RunSubprocessMocked(
+        side_effect=run_subprocess_side_effect(
+            (
+                (
+                    "/usr/sbin/grub2-mkconfig",
+                    "-o",
+                    "%s" % config_path,
+                ),
+                (
+                    "output",
+                    grub2_mkconfig_exit_code,
+                ),
+            ),
+            (("/usr/sbin/grub2-install", "/dev/sda"), ("output", grub2_install_exit_code)),
+        ),
+    )
+    monkeypatch.setattr(
+        utils,
+        "run_subprocess",
+        value=run_subprocess_mocked,
+    )
+
+    update_grub_instance.run()
+    if expected is not None:
+        assert expected in caplog.records[-1].message
+
+
+@pytest.mark.parametrize(
+    ("config_path", "grub2_mkconfig_exit_code", "grub2_install_exit_code", "expected"),
+    (
+        (
+            "/boot/grub2/grub.cfg",
+            1,
+            0,
+            set(
+                (
+                    actions.ActionMessage(
+                        level="WARNING",
+                        id="GRUB2_CONFIG_CREATION_FAILED",
+                        title="GRUB2 config file generation failed",
+                        description="The GRUB2 config file generation failed.",
+                        diagnosis=None,
+                        remediations=None,
+                    ),
+                )
+            ),
+        ),
+        (
+            "/boot/grub2/grub.cfg",
+            0,
+            1,
+            set(
+                (
+                    actions.ActionMessage(
+                        level="WARNING",
+                        id="GRUB2_INSTALL_FAILED",
+                        title="Couldn't install the new images with GRUB2",
+                        description="The new images could not be installed with GRUB2.",
+                        diagnosis=None,
+                        remediations=None,
+                    ),
+                )
+            ),
+        ),
+    ),
+)
+def test_update_grub_messages(
```

Comment:
```suggestion
def test_update_grub_action_messages(
```
Maybe I would add this to differ it from the standard logging messages. But not needed and it's just my feeling

Reply:
+1